Sources/MongoSwift/ReadWriteConcern.swift

```diff
@@ -115,5 +115,134 @@ public class ReadConcern: Equatable, CustomStringConvertible {
+public class WriteConcern: Equatable, CustomStringConvertible {
+
+    /// A pointer to a mongoc_write_concern_t
+    internal var _writeConcern: OpaquePointer?
+
+    /// Indicates whether to wait for the write operation to get committed to the journal.
+    public var journal: Bool? {
+        return mongoc_write_concern_get_journal(self._writeConcern)
+    }
+
+    /// Specifies the number of nodes that should acknowledge the write.
+    /// MUST be greater than or equal to 0. Only one of this and wTags may be set.
+    public var w: Int32? {
+        return mongoc_write_concern_get_w(self._writeConcern)
+    }
+
+    /// Indicates a tag for nodes that should acknowledge the write. Only one of this and w may be set.
+    public var wTag: String? {
+        guard let wTag = mongoc_write_concern_get_wtag(self._writeConcern) else { return nil }
+        return String(cString: wTag)
+    }
+
+    /// If the write concern is not satisfied within this timeout (in milliseconds),
+    /// the operation will return an error. The value MUST be greater than or equal to 0.
+    public var wtimeoutMS: Int32? {
+        return mongoc_write_concern_get_wtimeout(self._writeConcern)
+    }
+
+    /// Indicates whether this is an acknowledged write concern.
+    public var isAcknowledged: Bool {
+        return mongoc_write_concern_is_acknowledged(self._writeConcern)
+    }
+
+    /// Indicates whether this is the default write concern.
+    public var isDefault: Bool {
+        return mongoc_write_concern_is_default(self._writeConcern)
+    }
+
+    /// Indicates whether the combination of values set on this WriteConcern is valid.
+    public var isValid: Bool {
+        return mongoc_write_concern_is_valid(self._writeConcern)
+    }
+
+    private var asDocument: Document {
+        let doc = Document()
+        try? self.append(to: doc)
+        return doc
+    }
+
+    public var description: String {
+        return self.asDocument.description
+    }
+
+    /// Initializes a new, empty WriteConcern
+    public init() {
+        self._writeConcern = mongoc_write_concern_new()
+    }
+
+    /// Initializes a WriteConcern. Only one of w and wTag should be provided.
+    public init(journal: Bool? = nil, w: Int32? = nil, wTag: String? = nil, wtimeoutMS: Int32? = nil) {
+        self._writeConcern = mongoc_write_concern_new()
+        self.setJournal(journal)
+        self.setW(w)
+        self.setWTag(wTag)
+        self.setWTimeoutMS(wtimeoutMS)
+    }
+
+    /// Initializes a new WriteConcern by copying a mongoc_write_concern_t.
+    /// The caller is responsible for freeing the original mongoc_write_concern_t.
+    internal init(_ writeConcern: OpaquePointer?) {
+        self._writeConcern = mongoc_write_concern_copy(writeConcern)
+    }
+
+    /// Sets the journal value on this writeConcern
+    private func setJournal(_ journal: Bool?) {
+        if let journal = journal { mongoc_write_concern_set_journal(self._writeConcern, journal) }
```

Comment:
is it useful to have these as separately defined methods? Looks like they can just be embedded in the initializer, unless you want to make the public variables have getters/setters